unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -69,8 +73,8 @@ public UTF8String set(final String str) {
   /**
    * Updates the UTF8String with byte[], which should be encoded in UTF-8.
    */
-  public UTF8String set(final byte[] bytes) {
-    this.bytes = bytes;
+  protected UTF8String set(final byte[] bytes) {
+    this.bytes = (bytes != null) ? bytes : new byte[0];
```

Comment:
If we don't want to accept `null`, I'd prefer to throw an exception. It's much easier to find a bug later, if the `set` throws an exception instead of any other method that runs into a null pointer exception. What do you think?